pkg/util/aad/aad.go

```diff
@@ -74,6 +73,14 @@ func GetToken(ctx context.Context, log *logrus.Entry, clientID, clientSecret, te
 
 		return false, nil
 	}, timeoutCtx.Done())
+	if !done {
+		err = authorizer.RefreshWithContextRawError(ctx, log)
+		if err != nil {
+			err = api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidServicePrincipalToken, "properties.servicePrincipalProfile", "The provided service principal generated an invalid token.")
```

Comment:
Why `api.CloudErrorCodeInvalidServicePrincipalToken` and invalid token message?

Reply:
@m1kola : if error exists which means its a invalid/bad SP token

Reply:
I thought that we are trying to improve error propogation for the case where customer supplied invalid credentials. Hence `api.CloudErrorCodeInvalidServicePrincipalCredentials` should be used. Am I wrong?

Reply:
@m1kola : The scenario here is if the client mistakenly given wrong secret value they are getting generic error(Internal Server Error), which is hard to identify what is the mistake. They need a clear message so that they can fix the secret value from their end itself without asking for technical help.
We dont have much to do with error propogation in the existing logic.

Reply:
@m1kola : Yes you are right, it should be CloudErrorCodeInvalidServicePrincipalCredentials. will be updated in the final commit.